pkg/quarterdeck/apikeys.go

```diff
@@ -414,3 +415,33 @@ func (s *Server) APIKeyDelete(c *gin.Context) {
 
 	c.Status(http.StatusNoContent)
 }
+
+// APIKeyPermissions returns the API key permissions available to the user.
+func (s *Server) APIKeyPermissions(c *gin.Context) {
+	var (
+		err    error
+		claims *tokens.Claims
+	)
+
+	// Fetch the user claims from the request
+	if claims, err = middleware.GetClaims(c); err != nil {
+		c.Error(err)
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("user claims unavailable"))
+		return
+	}
+
+	// All users can create API keys with publisher and subscriber permissions.
+	out := []string{
+		perms.Publisher,
+		perms.Subscriber,
+	}
```

Comment:
One thing I was considering was adding descriptions to the response, although that would require a database lookup or duplicating them in code.

Reply:
I think a database lookup then filtering based on the user's permissions makes sense -- it's a bit more flexible if we want to add permissions in the future than having a hard coded list here and this method may include permissions that are not eligible for an API key.

In terms of performance; either the front-end or tenant can cache the responses, so it shouldn't overwhelm quarterdeck.

Reply:
I've added the database query, if you'd like to create a future story for adding descriptions, please feel free (might be a good user experience for the front-end).